src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,217 @@
+   */
+  private Object[] toObjectArray(Object value) {
+    if (value instanceof byte[]) {
+      return ArrayUtils.toObject((byte[]) value);
+    } else if (value instanceof short[]) {
+      return ArrayUtils.toObject((short[]) value);
+    } else if (value instanceof int[]) {
+      return ArrayUtils.toObject((int[]) value);
+    } else if (value instanceof long[]) {
+      return ArrayUtils.toObject((long[]) value);
+    } else if (value instanceof float[]) {
+      return ArrayUtils.toObject((float[]) value);
+    } else if (value instanceof double[]) {
+      return ArrayUtils.toObject((double[]) value);
+    } else if (value instanceof char[]) {
+      return ArrayUtils.toObject((char[]) value);
+    } else if (value instanceof boolean[]) {
+      return ArrayUtils.toObject((boolean[]) value);
+    } else {
+      return (Object[]) value;
+    }
+  }
+
+  /**
+   * Helper method that prints the components of the array
+   *
+   * @return String that represents the components of the array
+   */
+  private String printArrayComponents() {
+    String finalString = "{";
+    int length = Array.getLength(value);
+    if (value.getClass().getComponentType().isEnum()) {
+      for (int i = 0; i < length; i++) {
+        if (Array.get(value, i) == null) {
+          finalString += "null";
+        } else {
+          finalString += new EnumValue((Enum<?>) Array.get(value, i)).getValueName();
+        }
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    } else if (value.getClass().getComponentType() == Class.class) {
+      for (int i = 0; i < length; i++) {
+        if (Array.get(value, i) == null) {
+          finalString += "null";
+        } else {
+          finalString += ((Class<?>) Array.get(value, i)).getCanonicalName() + ".class";
+        }
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    } else {
+      for (int i = 0; i < length; i++) {
+        finalString += Value.toCodeString(Array.get(value, i));
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    }
+    finalString += "}";
+    return finalString;
+  }
+
+  @Override
+  public boolean evaluate(Object... objects) throws Throwable {
+    assert objects.length == 1;
+    return Arrays.equals(toObjectArray(value), toObjectArray(objects[0]));
+  }
+
+  @Override
+  public int getArity() {
+    return 1;
+  }
+
+  /** The arguments to which this contract can be applied. */
+  static TypeTuple inputTypes = new TypeTuple(Arrays.asList(JavaTypes.OBJECT_TYPE));
+
+  @Override
+  public TypeTuple getInputTypes() {
+    return inputTypes;
+  }
+
+  @Override
+  public String toCommentString() {
+    return null;
+  }
+
+  @Override
+  public String get_observer_str() {
+    return "ObserverEqArray";
+  }
+
+  @Override
+  public String toString() {
+    return "randoop.ObserverEqArray, value=" + StringsPlume.escapeJava(printArrayComponents());
```

Comment:
You don't need to quote this for appearance in Java source code.
Also, I would simplify this to "ObserverEqArary(" + printArrayComponents() + ")".